frontend/src/pages/article/article.tsx

```diff
@@ -1,23 +1,42 @@
-import { type FC } from 'react';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import { Layout } from '~/libs/components/components.js';
-import { useParams } from '~/libs/hooks/hooks.js';
+import { AppRoute } from '~/libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useParams,
+} from '~/libs/hooks/hooks.js';
 import { type TagType } from '~/libs/types/types.js';
+import { actions } from '~/slices/articles/articles.js';
 
 import { ArticleView } from './components/article-view/article-view.jsx';
 import { Author } from './components/author/author.js';
 import styles from './styles.module.scss';
 
-const Article: FC = () => {
+const Article: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
   const { id } = useParams();
 
-  const MOCKED_TEXT = `Envision this: there is a technology currently undergoing testing that, when released to the public, will become a long-awaited revolution in energy. This new technology promises to be safer and more efficient than anything we have on the market now. It  will affect that which we consider mundane — power tools, toys, laptops, smartphones —
-  and that which we consider exceptional — medical devices, spacecraft, and the innovative new vehicle designs needed to wean us off of fossil fuels. We have known about this  technology for centuries, yet until now we have only been able to take small steps towards its creation. Billions of dollars are pouring into research and billions more will be made once the
-  technology has been perfected and released.
+  if (!id) {
+    navigate(AppRoute.ARTICLES);
+  }
 
-  This description may sound a lot like that of fusion power. Yet it’s actually referring to the upcoming innovations in the realm of battery technology — specifically that of solid-state batteries. And while both fusion power and solid-state batteries have been labeled technologies of the future but never of today, advancements and investments in solid-state
-  materials have increased tremendously over the years. Today not only are there many major companies and credible researchers involved, it seems we may finally start seeing these batteries released in just the next few years.
-  What can we expect once this elusive, transformative technology is finally ready for mass production?`;
+  useEffect(() => {
+    void dispatch(actions.getArticle(String(id)));
+  }, [dispatch, id]);
+
+  const { article } = useAppSelector(({ articles }) => ({
+    article: articles.article,
+  }));
+
+  if (!article) {
```

Comment:
Maybe it would be better to set `article` to `null` in the initial state, when using an empty object this condition will never be true in case the app failed to fetch article